CMakeTests/AddCompileDefinitions.cmake

```diff
@@ -0,0 +1,42 @@
+# Add a C or C++ compile definitions to the current scope
+#
+# add_compile_definitions(def [def ...] [DEBUG | RELEASE])
+#
+# Can optionally add the definitions to Debug or Release configurations only, use this when
+# targeting multi-configuration generators like Visual Studio or Xcode.
+# Release configurations means NOT Debug, so it will work for RelWithDebInfo or MinSizeRel too.
+# The definitions are added to the COMPILE_DEFINITIONS folder property.
+# Supports generator expressions, unlike add_definitions()
+#
+# Examples:
+#   add_compile_definitions(FOO) -> -DFOO
+#   add_compile_definitions(_DEBUG DEBUG) -> -D_DEBUG
+#   add_compile_definitions(NDEBUG RELEASE) -> -DNDEBUG
+#   add_compile_definitions($<$<COMPILE_LANGUAGE:C>:THISISONLYFORC>)
+
+function(add_compile_definitions def)
+  set(defs)
+  list(APPEND defs ${def} ${ARGN})
+
+  list(GET defs -1 last_def)
+  if(last_def STREQUAL "DEBUG")
+    set(ADDDEF_DEBUG ON)
+    list(REMOVE_AT defs -1)
+  elseif(last_def STREQUAL "RELEASE")
+    set(ADDDEF_RELEASE ON)
+    list(REMOVE_AT defs -1)
+  endif()
+
+  set(config_compile_definitions)
+  foreach(def ${defs})
+    if(ADDDEF_DEBUG)
```

Comment:
Don't need to give each definition it's own configuration test, CMake supports lists embedded in generator expressions:

```cmake
if(ADDDEF_DEBUG)
    set(config_compile_definitions "$<$<CONFIG:Debug>:${defs}>")
elseif(ADDDEF_RELEASE)
    set(config_compile_definitions "$<$<NOT:$<CONFIG:Debug>>:${defs}>")
else()
    set(config_compile_definitions "${defs}")
endif()
```